syntax = "proto3";

service CommandExecutor {
  rpc ExecuteCommand (CommandRequest) returns (CommandResponse);
  rpc KillProcess (CommandRequest) returns (CommandResponse);
}

message CommandRequest {
    string executable = 1;
    string command = 2;
}

message CommandResponse {
  string status = 1;   // Response status
  string output = 2;   // Output of the executed command or result
}
